test/functional/wallet_orphanedreward.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test orphaned block rewards in the wallet."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+class OrphanedBlockRewardTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        # Generate some blocks and obtain some coins on node 0.  We send
+        # some balance to node 1, which will hold it as a single coin.
+        self.nodes[0].generate(150)
+        self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 10)
+        self.nodes[0].generate(1)
+
+        # Get a block reward with node 1 and remember the block so we can orphan
+        # it later.
+        self.sync_blocks()
+        blk = self.nodes[1].generate(1)[0]
+        self.sync_blocks()
+
+        # Let the block reward mature and send coins including both
+        # the existing balance and the block reward.
+        self.nodes[0].generate(150)
+        assert_equal(self.nodes[1].getbalance(), 10 + 25)
+        txid = self.nodes[1].sendtoaddress(address=self.nodes[0].getnewaddress(), amount=30, subtractfeefromamount=True)
+
+        # Orphan the block reward and make sure that the original coins
+        # from the wallet can still be spent.
+        self.nodes[0].invalidateblock(blk)
+        self.nodes[0].generate(200)
+        self.sync_blocks()
+        # Without the following abandontransaction call, the coins are
+        # not considered available yet.
+        assert_equal(self.nodes[1].getbalances()["mine"], {
+          "trusted": 0,
+          "untrusted_pending": 0,
+          "immature": 0,
+        })
+        # The following abandontransaction is necessary to make the later
+        # lines succeed, and probably should not be needed; see
+        # https://github.com/bitcoin/bitcoin/issues/14148.
+        self.nodes[1].abandontransaction(txid)
+        assert_equal(self.nodes[1].getbalances()["mine"], {
+          "trusted": 10,
+          "untrusted_pending": 0,
+          "immature": 0,
+        })
+        self.nodes[1].sendtoaddress(address=self.nodes[0].getnewaddress(), amount=9, subtractfeefromamount=True)
```

Comment:
Can this be removed?

Reply:
Yes it can, but I think having it in there is an extra check that the wallet does properly handle the coins.  So unless there is a strong preference to remove it, I'd rather keep it in to make the test more robust and descriptive.

Reply:
> ... having it in there is an extra check ...

I verified that if this `sendtoaddress` fails (which I forced by increasing the amount sent to 99999), the entire test fails, which is good. (I wasn't aware of that, which is why I thought this was an unnecessary line of code -- good to leave it in as you have. Need to put an RPC within a `try` to have its failure not fail the overall test.)